webhooks/virtualmachine/mutation/virtualmachine_mutator.go

```diff
@@ -310,6 +312,35 @@ func SetDefaultPowerState(
 	return false
 }
 
+// SetDefaultBiosUUID sets a default bios uuid for a new VM.
+// If CloudInit is the Bootstrap method, CloudInit InstanceID is also set to BiosUUID.
+// Return true if a default bios uuid was set, otherwise false.
+func SetDefaultBiosUUID(
+	ctx *context.WebhookRequestContext,
+	client client.Client,
+	vm *vmopv1.VirtualMachine) bool {
+
+	var wasMutated bool
+
+	if vm.Spec.BiosUUID == "" {
+		vm.Spec.BiosUUID = uuid.New().String()
```

Comment:
Can we make this New4 or New5? Just something that makes the version of the UUID explicit. That way we can ensure the validation is consistent with the expected version.

Reply:
I don't see those in the [uuid package](https://pkg.go.dev/github.com/google/uuid), just New, New6, New7. Should I add a helper/util func in vm-operator instead?

Reply:
Huh. Well, do we know what version of the UUID vSphere uses?

Reply:
FWIW, the checks in kube are just regexes https://github.com/kubernetes/kubernetes/blob/ac8ac0fc170eec7cdf292d841f8d026a65ce1681/vendor/github.com/go-openapi/strfmt/default.go#L61-L79

Reply:
According to @mikekim, VMware tends to prefer UUID4, actual "random" UUIDs. The Google UUID package lists the various types and what they are -- https://pkg.go.dev/github.com/google/uuid#New. It looks like `New` creates a UUID4, so we should be good with that.

Reply:
Agreed, added comments indicating we're using random UUID4 here.